core/schema/modulesource.go

```diff
@@ -147,19 +140,18 @@ type parsedRefString struct {
 	modPath    string
 	modVersion string
 	hasVersion bool
-	isGitHub   bool
 	kind       core.ModuleSourceKind
 }
 
 func parseRefString(refString string) parsedRefString {
 	var parsed parsedRefString
 	parsed.modPath, parsed.modVersion, parsed.hasVersion = strings.Cut(refString, "@")
-	parsed.isGitHub = strings.HasPrefix(parsed.modPath, "github.com/")
 
-	if !parsed.hasVersion && !parsed.isGitHub {
+	if _, err := os.Stat(refString); err == nil {
```

Comment:
From initial testing, I can successfully use dagger modules hosted on other VCS platforms like GitLab. However, I have broken `ModuleSourceKindLocal`. I have left this change here for discussion purposes.

Some examples, with debug output attached:

```sh
./bin/dagger call --debug -m gitlab.com/purpleclay/daggerverse/golang@v0.1.0 --src . mod-version
1.21.6
```

```sh
./bin/dagger call --debug -m gitlab.com/testing5740411/another/another2/another3/daggerverse/golang@v0.1.0 --src . mod-version
1.21.6
```

[nested-gitlab-groups.txt](https://github.com/dagger/dagger/files/14626918/nested-gitlab-groups.txt)


Looking at the current logic and how this function is used, I feel there must be a better way to detect a local path provided to the `-m` flag on the CLI. I completely understand I lack knowledge of how the Dagger engine works. And there might be an easy way of doing this, which isn't that obvious to me.

Reply:
I suspect currently if you had a local module in `./github.com/foo/bar` and called `-m github.com/foo/bar` that would probably break for the same reason. Maybe local modules should always be prefixed by `./` or absolute path?

Reply:
I was toying with using a convention like `/` or `./` to denote that the module is local, but omitting those is also a valid way of indicating a file path. Of course, it would be much better than making a remote module identifiable through a convention such as `https://`. The typing of those extra characters feels like a bad developer experience.

Could a distinction be made through the CLI itself? For example, could introducing a new flag for a local module be done? Of course, that would be a breaking change to the CLI.

I am happy to make the change you suggested for local module testing with the abovementioned prefixes to see how it looks and behaves.

Reply:
I agree requiring `https://` doesn't feel great. I think the lookup needs to happen both from the CLI directly and also during runtime for loading dependencies, so a different flag may not work anyway. Any ideas @jedevc @vito @sipsma @shykes ?